hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -443,43 +428,38 @@ private void throwEntityNotFoundIfDsDoesNotExist(final Long distributionSetID) {
     }
 
     @Override
-    public long countByFilters(final Collection<TargetUpdateStatus> status, final Boolean overdueState,
-            final String searchText, final Long installedOrAssignedDistributionSetId,
-            final Boolean selectTargetWithNoTag, final String... tagNames) {
-        final List<Specification<JpaTarget>> specList = buildSpecificationList(new FilterParams(status, overdueState,
-                searchText, installedOrAssignedDistributionSetId, selectTargetWithNoTag, tagNames));
-        return countByCriteriaAPI(specList);
+    public long countByFilters(final FilterParams filter) {
+        List<Specification<JpaTarget>> specifications = buildSpecificationList(filter);
+        if (CollectionUtils.isEmpty(specifications)) {
+            return targetRepository.count();
+        }
+        return targetRepository.count(SpecificationsBuilder.combineWithAnd(specifications));
     }
 
     private List<Specification<JpaTarget>> buildSpecificationList(final FilterParams filterParams) {
         final List<Specification<JpaTarget>> specList = new ArrayList<>();
-        if ((filterParams.getFilterByStatus() != null) && !filterParams.getFilterByStatus().isEmpty()) {
+
+        if (!filterParams.getFilterByStatus().isEmpty()) {
             specList.add(TargetSpecifications.hasTargetUpdateStatus(filterParams.getFilterByStatus()));
         }
-        if (filterParams.getOverdueState() != null) {
-            specList.add(TargetSpecifications.isOverdue(TimestampCalculator.calculateOverdueTimestamp()));
-        }
-        if (filterParams.getFilterByDistributionId() != null) {
-            throwEntityNotFoundIfDsDoesNotExist(filterParams.getFilterByDistributionId());
+        filterParams.getOverdueState().ifPresent(s -> //
```

Comment:
value of `s` might be `false`, should we still consider it to be an `overdue` state?

Reply:
maybe add a filter `.filter(state -> state == true)` before the `ifPresent` call

Reply:
Yes true. 
Done.